Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -66,6 +65,22 @@ bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) con
 
     batteryData.current = sections.at(2).toDouble();
 
+
+    if(!batteryData.time.isValid()){
+        return false;
+    }
+    if(batteryData.time.isNull()){
+        return false;
+    }
+    if(batteryData.voltage==NULL){
+        return false;
+    }
+    if(batteryData.current==NULL){
+        return false;
+    }
+
+
+
```

Comment:
the current and voltage won't be NULL even if it is incorrect, therefore this isn't the best way to debug the batdatafailure. Instead, read up on the toDouble documentation to see if you can approach this from a different angle :)